pkg/uroot/uroot.go

```diff
@@ -323,45 +325,15 @@ func (o *Opts) addSymlinkTo(logger ulog.Logger, archive *initramfs.Opts, command
 
 // resolvePackagePath finds import paths for a single import path or directory string
 func resolvePackagePath(logger ulog.Logger, env golang.Environ, pkg string) ([]string, error) {
-	// Search the current working directory, as well GOROOT and GOPATHs
-	prefixes := append([]string{""}, env.SrcDirs()...)
-	// Resolve file system paths to package import paths.
-	for _, prefix := range prefixes {
-		path := filepath.Join(prefix, pkg)
-		matches, err := filepath.Glob(path)
-		if len(matches) == 0 || err != nil {
-			continue
-		}
-
-		var importPaths []string
-		for _, match := range matches {
-
-			// Only match directories for building.
-			// Skip anything that is not a directory
-			fileInfo, _ := os.Stat(match)
-			if !fileInfo.IsDir() {
-				continue
-			}
-
-			p, err := env.PackageByPath(match)
-			if err != nil {
-				logger.Printf("Skipping package %q: %v", match, err)
-			} else if p.ImportPath == "." {
-				// TODO: I do not completely understand why
-				// this is triggered. This is only an issue
-				// while this function is run inside the
-				// process of a "go test".
-				importPaths = append(importPaths, pkg)
-			} else {
-				importPaths = append(importPaths, p.ImportPath)
-			}
-		}
-		return importPaths, nil
+	cfg := golang.DefaultConfig
+	pkgs, err := packages.Load(cfg, pkg)
```

Comment:
packages.Load is not quite right. You can see that I reimplemented a part of it in #1358 pkg/golang/build.go, because it doesn't quite respect the environment you're in. (I'm not sure that's still true today, a lot of changes have gone in in the last year.)

E.g. builds with GO111MODULE=off go1.13 should still work.